lib/backend/backend.go

```diff
@@ -106,6 +111,47 @@ type Backend interface {
 	CloseWatchers()
 }
 
+// newbk intializes a [Backend].
+type newbk interface {
+	new(context.Context, Params) (Backend, error)
+}
+
+// newbkfn is a function that can initialize a [Backend]. The function can return
+// any type that implements [Backend].
+type newbkfn[T Backend] func(context.Context, Params) (T, error)
+
+// new converts a generic backend type to the [Backend] interface. This allows
+// any backend intialization function to implement the newbk interface.
+func (fn newbkfn[T]) new(ctx context.Context, params Params) (Backend, error) {
+	return fn(ctx, params)
+}
+
+// MustRegister registers a [Backend] implementation. Panicking if it has already
+// been registered.
+func MustRegister[T Backend](fn newbkfn[T], types ...string) {
+	for _, t := range types {
+		if _, ok := newbks[t]; ok {
+			panic(fmt.Sprintf("backend already registered: %s", t))
+		}
+	}
+	for _, bkType := range types {
+		newbks[bkType] = fn
+	}
+}
+
+// Backend initializes a new [Backend] implementation based on the service config.
+func New(ctx context.Context, bkType string, params Params) (Backend, error) {
+	newbk, ok := newbks[bkType]
+	if !ok {
+		return nil, trace.BadParameter("unsupported secrets storage type: %q", bkType)
```

Comment:
```suggestion
		return nil, trace.BadParameter("unsupported storage type: %q", bkType)
```

To align with our teleport config structures